libs/jwst-storage/src/storage/docs/database.rs

```diff
@@ -71,142 +84,136 @@ impl DocDBStorage {
         .context("failed to insert update")?;
-        trace!("end insert: {table}");
+        trace!("end insert: {workspace}");
         Ok(())
     }
 
-    async fn replace_with<C>(conn: &C, table: &str, blob: Vec<u8>) -> JwstResult<()>
+    async fn replace_with<C>(conn: &C, workspace: &str, blob: Vec<u8>) -> JwstResult<()>
     where
         C: ConnectionTrait,
     {
-        trace!("start replace: {table}");
+        trace!("start replace: {workspace}");
         Docs::delete_many()
-            .filter(DocsColumn::Workspace.eq(table))
+            .filter(DocsColumn::Workspace.eq(workspace))
             .exec(conn)
             .await
             .context("failed to delete old updates")?;
         Docs::insert(DocsActiveModel {
-            workspace: Set(table.into()),
+            workspace: Set(workspace.into()),
             timestamp: Set(Utc::now().into()),
             blob: Set(blob),
             ..Default::default()
         })
         .exec(conn)
         .await
         .context("failed to insert new updates")?;
-        trace!("end replace: {table}");
+        trace!("end replace: {workspace}");
         Ok(())
     }
 
-    async fn drop<C>(conn: &C, table: &str) -> JwstResult<()>
+    async fn delete<C>(conn: &C, workspace: &str) -> JwstResult<()>
     where
         C: ConnectionTrait,
     {
-        trace!("start drop: {table}");
+        trace!("start drop: {workspace}");
         Docs::delete_many()
-            .filter(DocsColumn::Workspace.eq(table))
+            .filter(DocsColumn::Workspace.eq(workspace))
             .exec(conn)
             .await
             .context("failed to delete updates")?;
-        trace!("end drop: {table}");
+        trace!("end drop: {workspace}");
         Ok(())
     }
 
-    async fn update<C>(&self, conn: &C, table: &str, blob: Vec<u8>) -> JwstResult<()>
+    async fn update<C>(&self, conn: &C, workspace: &str, blob: Vec<u8>) -> JwstResult<()>
     where
         C: ConnectionTrait,
     {
-        trace!("start update: {table}");
-        let update_size = Self::count(conn, table).await?;
+        trace!("start update: {workspace}");
+        let update_size = Self::count(conn, workspace).await?;
         if update_size > MAX_TRIM_UPDATE_LIMIT - 1 {
-            trace!("full migrate update: {table}, {update_size}");
-            let data = Self::all(conn, table).await?;
+            trace!("full migrate update: {workspace}, {update_size}");
+            let doc_records = Self::all(conn, workspace).await?;
 
             let data = tokio::task::spawn_blocking(move || {
-                migrate_update(data, Doc::default())
-                    .and_then(|doc| {
-                        Ok(doc
-                            .transact()
-                            .encode_state_as_update_v1(&StateVector::default())?)
-                    })
-                    .expect("failed to encode update")
+                merge_doc_records(doc_records).expect("failed to encode update")
```

Comment:
consider exposing the result outside the `spawn_blocking`, using func()? syntax sugar process it